internal/evaluator/evaluator.go

```diff
@@ -282,3 +296,29 @@ func convertFS(apiFS *api.FileSystem) (fs fs.FileSystem, err error) {
 
 	return fs, err
 }
+
+func fsFromEnvironment(env map[string]string) (fs fs.FileSystem, err error) {
+	// Fallback to failing filesystem implementation that will fail the evaluation only
+	// if cirrus.fs is used
+	fs = failing.New(ErrGitHubFromEnvFailed)
+
+	// Use GitHub filesystem if all the required variables are present
+	owner, ok := env["CIRRUS_REPO_OWNER"]
+	if !ok {
+		return
+	}
+	repo, ok := env["CIRRUS_REPO_NAME"]
+	if !ok {
+		return
+	}
+	reference, ok := env["CIRRUS_CHANGE_IN_REPO"]
```

Comment:
This is not reflecting repositories' config resolution policy. It seems we need to pass an FS from the server side.